group-actors/src/main/java/org/sunbird/util/ActivitySearchRequestGenerator.java

```diff
@@ -0,0 +1,136 @@
+package org.sunbird.util;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sunbird.models.ActivitySearchRequestConfig;
+import org.sunbird.models.SearchRequest;
+import org.sunbird.service.impl.SearchActivityServiceImpl;
+import org.sunbird.util.helper.PropertiesCache;
+
+public class ActivitySearchRequestGenerator {
+
```

Comment:
@sknirmalkar89 - i think this is not generic enough to support all types of activities. Also, the logic seems to be complex and this configuration/code will become even more complex if we go this way to generalise the activity details fetch operation.